test/common/testvm.py

```diff
@@ -96,23 +91,50 @@ def __str__(self):
 class RepeatableFailure(Failure):
     pass
 
+LOGIN_MESSAGE = """
+TTY LOGIN
+  User: {ssh_user}/admin  Password: foobar
+  To quit use Ctrl+], Ctrl+5 (depending on locale)
+
+SSH ACCESS
+  $ ssh -p {ssh_port} {ssh_user}@{ssh_address}
+  Password: foobar
+
+COCKPIT
+  http://{web_address}:{web_port}
+"""
+
 class Machine:
-    def __init__(self, address=None, image=None, verbose=False, label=None, fetch=True):
+    def __init__(self, address=None, image=None, verbose=False, label=None, browser=None):
         self.verbose = verbose
 
-        # Currently all images are x86_64. When that changes we will have
-        # an override file for those images that are not
-        self.arch = "x86_64"
-
         self.image = image or testinfra.DEFAULT_IMAGE
         self.atomic_image = self.image in testinfra.ATOMIC_IMAGES
-        self.fetch = fetch
-        self.vm_username = "root"
-        self.address = address
-        self.label = label or "UNKNOWN"
+        self.ssh_user = "root"
+        if ":" in address:
+            (self.ssh_address, unused, self.ssh_port) = address.rpartition(":")
+        else:
+            self.ssh_address = address
+            self.ssh_port = 22
+        if browser and ":" in browser:
+            (self.web_address, unused, self.web_port) = browser.rpartition(":")
+        else:
+            self.web_address = address
+            self.web_port = browser or 9090
```

Comment:
It's there see VirtNetwork and VirtMachine. I've updated it to make it consistent.